core/org.eclipse.cdt.core/utils/org/eclipse/cdt/utils/DefaultGnuToolFactory.java

```diff
@@ -161,4 +165,28 @@ protected String[] getEnvironment() {
 		return Arrays.stream(vars).map(v -> String.format("%s=%s", v.getName(), v.getValue())) //$NON-NLS-1$
 				.toArray(String[]::new);
 	}
+
+	/** @since 8.2 */
+	protected String getToolPrefix() {
+		ICConfigExtensionReference ref = fExtension.getConfigExtensionReference();
+		ICConfigurationDescription cfg = ref.getConfiguration();
+		ICdtVariable[] userVars = CCorePlugin.getUserVarSupplier().getMacros(cfg);
+		ICdtVariable var = Arrays.stream(userVars).filter(v -> v.getName().equals(GNU_TOOL_PREFIX_VARIABLE)).findFirst()
+				.orElse(null);
+
+		// if user-defined variable not found, look for system variable provided by toolchain integration
+		if (null == var) {
```

Comment:
I'm not against using the "null == var" syntax rather than "var == null", but it looks like the file are using the latter way of writing the expression rather than what you used.
Do you think we should continue with the "var == null" order throughout the file or have a mix?
I'm okay regardless what you end up with!

Reply:
I use `null == var`  habitually, but I don't think this is worth arguing over. Will change for consistency.

Reply:
Fixed.